### PR TITLE
Azcopy sync option for file share is not supported

### PR DESCRIPTION
**Pull request by dphulkar-msft**
**Issue Link:** https://github.com/Azure/azure-storage-azcopy/issues/2650

**Issue:**
The customer found the data loss issues when they used "Azcopy sync" to sync their on-premises files (millions of files) to Azure files. They detected few thousands of files are corrupted. 
Azcopy sync operation is not supported and is already documented [here](https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-files#synchronize-files) 

**Resolution:**
Added a warning message which will be displayed on the console and in the job logs as well if the user tries to perform sync operation for file share.

**Console Output:**
Screenshot 2024-05-02 163228

**Job Log Output:**
[32b9249b-c4d7-9947-4352-1d7011f7eba9.log](https://github.com/Azure/azure-storage-azcopy/files/15187272/32b9249b-c4d7-9947-4352-1d7011f7eba9.log)

### CODE REVIEW

**Review by pranavmalik-msft — COMMENTED** [review]
You may want to update your branch to sync the latest changes.

File: cmd/zt_sync_local_blob_test.go [hunk 1]
```diff
@@ -468,3 +469,53 @@ func TestDryrunSyncLocaltoBlob(t *testing.T) {
 		a.True(testDryrunStatements(blobsToDelete, msg))
 	})
 }
+
+// TestSyncGenerateWarningForLocalToFSUpload
+// @brief This tests performs sync operation from local to file share or vice versa.
+// @validation This test verifies if the warning message is displayed on console if the user tries to perform sync operation from local to file share.
+func TestSyncGenerateWarningForLocalToFSUpload(t *testing.T) {
```
Comment (gapra-msft) [inline cmd/zt_sync_local_blob_test.go:476]: Can you add this to the new e2e test framework instead of as a unit test?

Comment (dphulkar-msft) [reply cmd/zt_sync_local_blob_test.go]: Added to e2e test framework,

File: cmd/zt_sync_local_blob_test.go [hunk 2]
```diff
@@ -22,14 +22,15 @@ package cmd
...
 	"context"
-	"github.com/stretchr/testify/assert"
 	"os"
 	"path/filepath"
 	"sort"
 	"strings"
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
```
Comment (gapra-msft) [inline cmd/zt_sync_local_blob_test.go:32]: remove these ut changes

Comment (dphulkar-msft) [reply cmd/zt_sync_local_blob_test.go]: done

**Review by pranavmalik-msft — APPROVED** [review]
LGTM

**Review by vibhansa-msft — APPROVED** [review]

File: cmd/zt_sync_comparator_test.go [hunk 3]
```diff
@@ -619,10 +620,13 @@ func TestDryrunSyncLocaltoFile(t *testing.T) {
 		for i := 0; i < len(msg); i++ {
 			if strings.Contains(msg[i], "DRYRUN: remove") {
 				a.True(strings.Contains(msg[i], dstShareClient.URL()))
-			} else {
+			} else if strings.Contains(msg[i], "DRYRUN: copy") {
```
Comment (gapra-msft) [inline cmd/zt_sync_comparator_test.go:623]: are these changes necessary? Otherwise this PR looks good to me.

**Review by gapra-msft — APPROVED** [review]

File: cmd/zt_sync_comparator_test.go (hunk 3 above)
Comment (dphulkar-msft) [reply cmd/zt_sync_comparator_test.go:623]: Yes, we can keep this because now we have added a warn message as part of verification. In this loop we are verifying all the expected logs while performing sync operation from local to file share.
